src/swap/SwapReviewScreen.tsx

```diff
@@ -193,11 +225,8 @@ export function SwapReviewScreen() {
                 <View style={styles.tokenDisplayView}>
                   <TokenDisplay
                     style={[styles.amountText, { color: colors.greenUI }]}
-                    // Fix: I think we shouldn't subtract gas from the buyAmount here.
                     amount={divideByWei(
```

Comment:
This isn't right for tokens not using 18 decimals, is this the bug you talked about on Slack?

Reply:
I saw a similar issue in other places, I'll add this case to the [task](https://linear.app/valora/issue/RET-712/[wallet]-swapping-tokens-with-different-decimals-is-not-working).